doomworld_downloader/doomworld_downloader.py

```diff
@@ -71,21 +82,43 @@ def parse_thread_list(page_number):
     threads = []
     for thread in thread_elems:
         id = thread['data-rowid']
+        if id in THREAD_MAP_KEYED_ON_ID:
+            if THREAD_MAP_KEYED_ON_ID[id].get('additional_info', {}).get('ignore', False):
+                continue
         title = thread.find(class_='ipsDataItem_title')
         title_link = title.find_all('a')[0]
+        pagination = title.find(class_='ipsPagination')
+        if pagination is not None:
+            last_page_num = int(pagination.getText().strip().split()[-1])
+        else:
+            last_page_num = 1
+
         last_poster = thread.find(class_='ipsDataItem_lastPoster')
         last_post_date = last_poster.find('time')['datetime']
         last_post_date = datetime.strptime(last_post_date, '%Y-%m-%dT%H:%M:%SZ')
-        threads.append(Thread(title_link.getText().strip(), id, title_link['href'], last_post_date))
+        threads.append(Thread(title_link.getText().strip(), int(id), title_link['href'],
+                              last_post_date, last_page_num))
 
     return threads
 
 
-def parse_thread_page(base_url, page_number):
+def parse_thread_page(base_url, page_number, thread):
     soup = get_page(THREAD_URL.format(base_url=base_url, num=page_number))
     post_elems = soup.find_all('article', class_='ipsComment')
     posts = []
     for post in post_elems:
+        post_content_elem = post.find('div', class_='cPost_contentWrap')
+        post_content_elem = post_content_elem.find('div', attrs={'data-role': 'commentContent'})
+        attachments = get_link_elems(post_content_elem.find_all('a', class_='ipsAttachLink'),
+                                     extract_link=True)
+        # Skip posts with no attachments as they have no demos to search for
+        if not attachments:
```

Comment:
You could also skip posts whose attachments do not end in ".zip".
I recall people who uploaded .rar files, or directly .lmp files, which are not eligible to be uploaded to DSDA.

You can leave this for a future development of course, to separate stuff :)

Reply:
Sounds good, I'll skip them for now. Will add a TODO to repackage other compressed formats I think as well.